draft5/app/controllers/ValidatorController.js

```diff
@@ -133,4 +144,17 @@ app.controller('validatorController', function ($scope, $http, $window) {
     self.validateDocument();
   });
 
+  // Save form data before reload
+  $window['onbeforeunload'] = function () {
```

Comment:
Ah, sorry, I totally misunderstood your question. Yes that works. Pushing the change now.